modules/misc/debug.nix

```diff
@@ -0,0 +1,27 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+{
+  options.home = {
+    enableDebugInfo = mkEnableOption "debug info" // {
+      description = ''
+        Some Nix(OS)-packages provide debug-symbols for <command>gdb</command>
+        in the <literal>debug</literal>-output. This option ensures that those are
+        automatically fetched from the binary cache if available and <command>gdb</command>
+        is configured to find those symbols.
+      '';
+    };
+  };
+
+  config = mkIf config.home.enableDebugInfo {
+    home.extraOutputsToInstall = [ "debug" ];
+
+    home.sessionVariables = {
+      NIX_DEBUG_INFO_DIRS = [
```

Comment:
Wouldn't you need something like
```
NIX_DEBUG_INFO_DIRS =
  "$NIX_DEBUG_INFO_DIRS"
  + "${$NIX_DEBUG_INFO_DIRS:+:}"
  + "${{config.home.profileDirectory}/lib/debug"
```
here to guard against an empty `$NIX_DEBUG_INFO_DIRS`?

Also note the use of `${config.home.profileDirectory}` to cope with the case where the profile ends up in `/etc/profiles/per-user`.

Reply:
Oof you're right. I only considered the case where `environment.enableDebugInfo` is set to `true` - sorry!. Will fix it later today :)

Reply:
Applied your suggestions and tested it locally. Thanks!